grails-app/views/layouts/_heatmap.gsp

```diff
@@ -88,8 +88,12 @@
                         script-to-run="run"
                         arguments-to-use="runParams"
                         serialized="true"></run-button>
-            <capture-plot-button filename="heatmap.svg"></capture-plot-button>
-            <download-results-button></download-results-button>
+            <capture-plot-button filename="heatmap.svg" disabled="scriptResults.disableDownloadButtons">
+
+            </capture-plot-button>
+            <download-results-button disabled="scriptResults.disableDownloadButtons">
```

Comment:
I see what you are trying to do, but you are creating some kind of global variable within the results object.